### PR TITLE
[luci] weight quantized-dequantized module about channel-wise int16

### PR DESCRIPTION
**Pull request by meejeong**
- implement to choose Convolution operators with quantize & dequantize
- quantize & dequantize weights per channel
- datatype : int16
- symmetric quantization

Related to: #696, https://github.sec.samsung.net/STAR/nnapps/issues/379

ONE-DCO-1.0-Signed-off-by: mia park <mia.park@samsung.com>

### CODE REVIEW

File: compiler/luci/pass/src/QuantizeDequantizeWeightsPass.cpp [hunk 1]
```diff
@@ -29,6 +29,33 @@ namespace luci
 
 namespace
 {
+void compute_sym_scale_zp(float min, float max, float &scaling_factor, int64_t &zp,
+                          float &nudged_min, float &nudged_max)
+{
+  assert(min != max);
+
+  const int32_t kMinScale = std::numeric_limits<int16_t>::lowest() + 1;
+  const int32_t kMaxScale = std::numeric_limits<int16_t>::max();
```
Comment (mhs4670go) [inline compiler/luci/pass/src/QuantizeDequantizeWeightsPass.cpp:38]: Just curiosity, 
1. why does `kMinScale` need `+1`?
2. Maybe `std::numeric_limits<int32_t>` looks right not `int16_t`, doesn't it?

Comment (jinevening) [reply compiler/luci/pass/src/QuantizeDequantizeWeightsPass.cpp:38]: 1. For symmetric quantization, the range of quantized values should be symmetric (-x ~ x). The lowest() of int16 is -32768 and max() is 32767. So `kMinScale` needs +1.
2. This is for int16 quantization, so `int16_t` would be right.

File: compiler/luci/pass/src/QuantizeDequantizeWeightsPass.cpp [hunk 2]
```diff
@@ -29,6 +29,33 @@ namespace luci
 
 namespace
 {
+void compute_sym_scale_zp(float min, float max, float &scaling_factor, int64_t &zp,
```
Comment (seanshpark) [inline compiler/luci/pass/src/QuantizeDequantizeWeightsPass.cpp:32]: please add an empty space between `{` and `void` line

File: compiler/luci/pass/src/QuantizeDequantizeWeightsPass.cpp (hunk 1 above)
Comment (seanshpark) [reply compiler/luci/pass/src/QuantizeDequantizeWeightsPass.cpp:38]: (optional)  maybe
```cxx
const int32_t kMinScale  = -kMaxScale
```

File: compiler/luci/pass/src/QuantizeDequantizeWeightsPass.cpp [hunk 3]
```diff
@@ -180,6 +207,102 @@ void cal_minmax_per_channel(CircleConst *node, std::vector<float> &min, std::vec
   }
 }
 
+void sym_wquant_per_channel(CircleConst *node, std::vector<float> &min, std::vector<float> &max,
+                            std::vector<float> &scaling_factor, std::vector<int64_t> &zp,
+                            std::vector<float> &nudged_min, std::vector<float> &nudged_max)
+{
+  assert(node->dtype() == loco::DataType::FLOAT32);
+  const int32_t kMinScale = std::numeric_limits<int16_t>::lowest() + 1;
+  const int32_t kMaxScale = std::numeric_limits<int16_t>::max();
```
Comment (seanshpark) [inline compiler/luci/pass/src/QuantizeDequantizeWeightsPass.cpp:216]: (optional) same here

File: compiler/luci/pass/src/QuantizeDequantizeWeightsPass.cpp [hunk 4]
```diff
@@ -180,6 +207,102 @@ void cal_minmax_per_channel(CircleConst *node, std::vector<float> &min, std::vec
...
+
+  loco::TensorShape dimension;
+  dimension.rank(4);
+  uint32_t indices[4] = {
+      0,
+  };
+  int channel_dim_index{0};
+
+  bool ret = get_channel_dim_index(node, dimension, channel_dim_index);
+  assert(ret);
```
Comment (seanshpark) [inline compiler/luci/pass/src/QuantizeDequantizeWeightsPass.cpp:234]: `ret` will fail as unused var in release build. please fix this as previous fix I've made.

File: compiler/luci/pass/src/QuantizeDequantizeWeightsPass.cpp [hunk 5]
```diff
@@ -180,6 +207,102 @@ void cal_minmax_per_channel(CircleConst *node, std::vector<float> &min, std::vec
...
```
Comment (seanshpark) [inline compiler/luci/pass/src/QuantizeDequantizeWeightsPass.cpp:279]: same here

File: compiler/luci/pass/src/QuantizeDequantizeWeightsPass.cpp (hunk 1 above)
Comment (meejeong) [reply compiler/luci/pass/src/QuantizeDequantizeWeightsPass.cpp:38]: I will fix it based on @seanshpark 's comment

**Pull request merged**